extensions/wikia/TemplateClassification/scripts/TemplateClassificationModal.js

```diff
@@ -201,13 +203,42 @@ function ($, w, mw, loader, nirvana, tracker, throbber, labeling) {
 			});
 		}
 
-		if (modalMode === 'addTypeBeforePublish' && newTemplateType) {
+		if (modalMode === forceClassificationModalMode && newTemplateType) {
 			$('#wpSave').click();
+		} else if (shouldRedirectToInfoboxBuilder(newTemplateType, modalMode)) {
+			throbber.show(modalInstance.$content);
+			redirectToInfoboxBuilder(mw.config.get('wgTitle'));
 		} else {
 			modalInstance.trigger('close');
 		}
 	}
 
+	/**
+	 * @desc checks if infobox builder should be used instead of regular editor
+	 * @param {String} newTemplateType - choosen template type
+	 * @param {String} modalMode - mode in which modal was opened
+	 * @returns {Boolean}
+	 */
+	function shouldRedirectToInfoboxBuilder(newTemplateType, modalMode) {
+		return w.isSupportedInfoboxTemplateBody &&
+			newTemplateType === 'infobox' &&
+			modalMode !== forceClassificationModalMode;
+	}
+
+	/**
+	 * @desc redirects to infobox builder tool
+	 */
+	function redirectToInfoboxBuilder() {
+		w.location = w.infoboxBuilderPath;
```

Comment:
Checked in https://github.com/Wikia/app/pull/9865/files#diff-d5bde9afc6a40ff66cf72bf44a3fc4b2R192